src/agentNet/agent_net.py

```diff
@@ -0,0 +1,83 @@
+import networkx as nx
+import random
+import matplotlib.pyplot as plt
+import yaml
+from network_utils import *
+
+class AgentNet(nx.Graph):
+    """
+    Class with agent network graph.
+    """
+    def __init__(self, init_num=2, net_type="complete", net_density="MEDIUM"):
+        """
+        Method initializes an agent network with default parameters.
+        :param init_num: initial number of nodes for random graph construction algorithms or initial nodes for complete graph
+        :param net_type: type of graph {"complete", "random", "custom"}
+        :param net_density: density of random graph {SPARSE, MEDIUM, DENSE}
+        """
+        self.init_num =  init_num 
+        self.net_type = net_type 
+        self.net_density = net_density 
+    def create_network(self):
+        """
+        Method creates an agent network up to net_type.
+        """
+        if self.net_type == "complete":
+            self.network = nx.complete_graph(n = self.init_num)
+        elif self.net_type == "custom":
+            self.network = nx.Graph(self.connections)
+        else:
+            self.network = nx.fast_gnp_random_graph(self.init_num, dens2prob(self.net_density))
+    def set_params(self, config_filename):
+        """
+        Method sets randomly a file link to agent policy for each agent id. 
+        :param config_filename: path to file with links to policies
+        """
+        with open(config_filename, 'r') as configs:
+            params_yaml = yaml.load(configs)
+            self.agents_policies = {agent_id: random.choices(params_yaml.get('configs')) for agent_id in self.network.nodes}
+    def add_agent(self, agent_id):
```

Comment:
add_agent nie powinno dodawać losowego połączenia domyślnie, może to być opcjonalnie albo dla `net_type="random"`